api/app/Policies/PoolCandidatePolicy.php

```diff
@@ -51,68 +75,123 @@ public function create(User $user)
      * @param  \App\Models\PoolCandidate  $poolCandidate
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function update(User $user, PoolCandidate $poolCandidate)
+    public function update(?User $user, PoolCandidate $poolCandidate)
     {
-        return $user->isAdmin();
+        // Only update an owner can do is submission, so check that
+        if ($user?->id === $poolCandidate->user_id) {
+            $user?->isAbleTo("submit-own-application");
+        }
+
+        $candidatePoolTeam = $this->userBelongsToPoolCandidatePool($user, $poolCandidate);
+
+        if ($candidatePoolTeam) {
+            return $user?->isAbleTo("update-team-applicationStatus", $candidatePoolTeam, true);
+        }
+
+        return false;
     }
 
     /**
-     * Determine whether the user can delete the model.
+     * Determine whether the user can submit the model.
      *
      * @param  \App\Models\User  $user
      * @param  \App\Models\PoolCandidate  $poolCandidate
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function delete(User $user, PoolCandidate $poolCandidate)
+    public function submit(?User $user, PoolCandidate $poolCandidate)
     {
-        return $user->isAdmin();
+        return  $user?->id === $poolCandidate->user_id && $user?->isAbleTo("submit-own-application");
     }
 
     /**
-     * Determine whether the user can restore the model.
+     * Determine whether the user can archive the model.
      *
      * @param  \App\Models\User  $user
      * @param  \App\Models\PoolCandidate  $poolCandidate
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function restore(User $user, PoolCandidate $poolCandidate)
+    public function archive(?User $user, PoolCandidate $poolCandidate)
     {
-        return $user->isAdmin();
+        return  $user?->id === $poolCandidate->user_id && $user?->isAbleTo("archive-own-submittedApplication");
```

Comment:
So, I thought about doing this but I don't think a policy is the correct place to do this. Not opposed to doing it here but I thought validation would be a more appropriate place to check that.

Reply:
I wondered about this as well - whether this would be more appropriate in a validator. I'm currently leaning to putting it here in the cases where we've tied the permission to a specific state - can we really say the `archive-own-submittedApplication` permission is applicable if we don't know if its a submitted application? I'm open to more discussion about this.

Reply:
Actually, it should be in both because both can be true and should be guarded against. Validation allows us to send a better error message related to the data but we also only want to authorize it if it is submitted.

1. A draft application cannot be archived so it is not valid
2. A user is only authorized to archive submitted applications

Reply:
Sorry to take you on this wild ride 🎢  but...I had to back that all out. So, policies run before validation because I guess what is the point validating the data if you aren't authorized to complete the action. So what happens is:

1. Check policy
2. See application is not submitted
3. Exits early with 401

So, we never get the proper, nice error message for the user not being able to archive/suspend/delete the application and why and instead just see "unauthorized".

[fix suspend, archive, delete policy/tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5815/commits/87e0241fc8f943bca7e20924a718c80f565b6bd5)

Reply:
Sorry to take you on this wild ride 🎢  but...I had to back that all out. So, policies run before validation because I guess what is the point validating the data if you aren't authorized to complete the action. So what happens is:

1. Check policy
2. See application is not submitted
3. Exits early with 401

So, we never get the proper, nice error message for the user not being able to archive/suspend/delete the application and why and instead just see "unauthorized".

[fix suspend, archive, delete policy/tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5815/commits/87e0241fc8f943bca7e20924a718c80f565b6bd5)

Reply:
Yeah, kinda sad we can't do it in both places 😢 

Added documentation in [add docs, check not draft for submitted permission](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5815/commits/78d1f2c7a0b2b28763f02aa610063f526e6e4bb7)